replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,259 @@
+
+    if (infos.size() > 1) {
+      task.fail(ErrorCode.UNSUPPORTED_ERROR,
+          "more than one metadata is currently not supported by this worker");
+      return;
+    }
+
+    if(infos.isEmpty()) {
+      task.fail(ErrorCode.UNSUPPORTED_ERROR,
+          "unable to handle any of the metadatas received in this task");
+      return;
+    }
+
+    final DDFMetadataInfo info = infos.get(0);
+    final Metadata metadata = new MetadataImpl(((Map) info.getData()).get("raw-metadata"),
+        info.getDataClass(), task.getId(),
+        Date.from(info.getLastModified()));
+
+    final OperationType type = task.getOperation();
+    switch (type) {
+      case HARVEST:
+      case REPLICATE_FROM:
+        doCreate(task, metadata, destination, local);
+        break;
+      case REPLICATE_TO:
+        doCreate(task, metadata, local, destination);
+        break;
+      case DELETE:
+        // always delete on non-local site
+        doDelete(task, metadata);
+        break;
+      default:
+        // should never hit this since deserialization will return UNKNOWN
+        task.fail(ErrorCode.UNSUPPORTED_ERROR,
+            String.format("Operation task %s is not supported by this worker", task));
+    }
+  }
+
+  private void doCreate(final Task task, final Metadata metadata, final NodeAdapter source,
+      final NodeAdapter destination)
+      throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+    if (source.exists(metadata)) {
+      CreateRequest request = new CreateRequestImpl(List.of(metadata));
+      boolean success = destination.createRequest(request);
+      if (!success) {
+        task.fail(ErrorCode.OPERATION_FAILURE,
+            String.format("Failed to %s metadata %s from %s to %s",
+                operation,
+                metadataId,
+                source.getSystemName(),
+                destination.getSystemName()));
+      }
+    } else {
+      LOGGER.info("Received task to create metadata on site {}, but it was deleted on {}",
+          destination.getSystemName(), destination.getSystemName());
+      task.complete();
+    }
+
+    ResourceInfo resourceInfo = task.getResource().orElse(null);
+    if (resourceInfo != null) {
+      URI uri = resourceInfo.getResourceUri().orElse(null);
+      if (uri != null) {
+        ResourceRequest request = new ResourceRequestImpl(metadata);
+        try {
+          ResourceResponse response = source.readResource(request);
+          Resource resource = response.getResource();
+          boolean success = destination
+              .createResource(new CreateStorageRequestImpl(List.of(resource)));
+          if (!success) {
+            task.fail(ErrorCode.OPERATION_FAILURE,
+                String.format("Failed to %s resource %s from %s to %s",
+                    operation,
+                    uri,
+                    source.getSystemName(),
+                    destination.getSystemName()));
```

Comment:
❗️ This will fall-through to the `task.complete()` below and fail. You have to return